src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1519,9 +2411,27 @@ public void setContextClassLoader(ClassLoader cl) {
         if (sm != null) {
             sm.checkPermission(new RuntimePermission("setContextClassLoader"));
         }
+        if (!isSupportedClassLoader(contextClassLoader)) {
```

Comment:
For a few moments, I thought this was a typo and should instead have been:
```
if (!isSupportedClassLoader(cl)) {
```
but then thinking a bit more, I realized what's being done here. Essentially, this checks whether the _current_ context classloader  is set to "not allowed to change/set context classloader" (represented by `Constants.NOT_SUPPORTED_CLASSLOADER`).

Do you think this line deserves some short code comment that it intentionally passes the current contextClassLoader (and not the one passed to the method) to `isSupportedClassLoader`?

Reply:
I think the comment on isSupportedClassLoader is readable so I think it's okay.